tests/unit/lib/SGMatrix_unittest.cc

```diff
@@ -761,3 +761,31 @@ TEST(SGMatrixTest,iterator)
 	for (auto v: mat)
 		EXPECT_EQ(mat[index++], v);
 }
+
+TEST(SGMatrixTest,column_iterator)
+{
+	constexpr index_t size=5;
+	SGMatrix<float64_t> mat(size, size);
+	linalg::range_fill(mat, 1.0);
+
+	SGMatrix<float64_t>::column_iterator col_itr(mat);
+
+	EXPECT_EQ(mat.get_column(0),*col_itr++);
+	++col_itr;
+	EXPECT_EQ(mat.get_column(2),*col_itr);
+	--col_itr;
+	EXPECT_EQ(mat.get_column(1),*col_itr);
+	col_itr += 2;
+	EXPECT_EQ(mat.get_column(3),*col_itr);
+	col_itr -= 1;
+	EXPECT_EQ(mat.get_column(2),*col_itr);
+	EXPECT_EQ(mat.get_column(3),col_itr[1]);
+	
+	auto new_itr = col_itr + 2;
+	EXPECT_EQ(mat.get_column(4),*new_itr);
+
+	//range-based loop should work as well
+	int idx;
+	std::for_each(mat.begin_column(),mat.end_column(),[&idx](auto* vec){ EXPECT_EQ(*vec,mat.get_column(idx++));});
```

Comment:
Imo std::for_each is sufficient. The tests above are a bit convoluted..

Reply:
I'll fix that.

Reply:
Thanks :)

Reply:
Have a check on them, see if it can be improved?